e2e/test/helpers/X509Certificate2Helper.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using FluentAssertions;
+
+namespace Microsoft.Azure.Devices.E2ETests.Helpers
+{
+    internal static class X509Certificate2Helper
+    {
+        internal static void GenerateSelfSignedCertificateFiles(string subject, string certificatePassword, DirectoryInfo destinationCertificateFolder, MsTestLogger logger)
+        {
+            GenerateSignedCertificateFiles(subject, certificatePassword, null, destinationCertificateFolder, logger);
+        }
+
+        internal static void GenerateIntermediateCertificateSignedCertificateFiles(
+            string leafCertificateSubject,
+            string certificatePassword,
+            string intermediateCertificateSubject,
+            DirectoryInfo destinationCertificateFolder,
+            MsTestLogger logger)
+        {
+            GenerateSignedCertificateFiles(leafCertificateSubject, certificatePassword, intermediateCertificateSubject, destinationCertificateFolder, logger);
+        }
+
+        private static void GenerateSignedCertificateFiles(
+            string leafCertificateSubject,
+            string certificatePassword,
+            string intermediateCertificateSubject,
+            DirectoryInfo destinationCertificateFolder,
+            MsTestLogger logger)
+        {
+            // Generate the private key for the self-signed certificate
+            logger.Trace($"Generating the private key for the self-signed certificate with subject {leafCertificateSubject} using ...\n");
+            string keyGen = $"genpkey" +
+                $" -out \"{destinationCertificateFolder}\\{leafCertificateSubject}.key\"" +
+                $" -algorithm RSA" +
+                $" -pkeyopt rsa_keygen_bits:2048";
+
+            logger.Trace($"openssl {keyGen}\n");
```

Comment:
add a comment calling out why openssl was chosen to generate certificates (lack of .NET native API support/ use bouncycastle)